torch/distributed/rpc/__init__.py

```diff
@@ -165,6 +166,10 @@ def init_rpc(
             # If world_size is not set in construction and also not set in environment variables
             # The store will be created for the dynamic group setting
             store = dist._create_store_from_options(rpc_backend_options, rank)
+            # If there is unspecified rank or rank is pre-allocated 0 (master), then rank will be allocated by store
+            if rank <= 0:
+                val = store.add(STORE_GET_RANK_KEY, 1)
+                rank = val - 1
```

Comment:
for rank > 0, what happens if it is duplicated with an auto-allocated rank? Say I do the following

init_rpc("A")
int_rpc("B")
int_rpc("C", rank=1)